src/serlio/materials/StingrayMaterialNode.cpp

```diff
@@ -68,6 +68,10 @@ void setTexture(MELScriptBuilder& sb, const std::wstring& target, const std::wst
 		sb.setAttr(MEL_VAR_MAP_NODE, L"fileTextureName", MEL_VAR_MAP_FILE);
 
 		sb.connectAttr(MEL_VAR_MAP_NODE, L"outColor", MEL_VAR_SHADER_NODE, L"TEX_" + target);
+
+		if (target.compare(L"opacity_map") == 0)

```

Comment:
We should make this generic. Consider pulling the check into the call site and by passing a string `alphaTarget` (empty by default = do not connect). This way we do not need to hard code knowledge about `opacity_map` and `opacity_map_uses_alpha_channel` in this helper function.

Reply:
fixed.